src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/github/v3/RestApplicationAuthenticationProvider.java

```diff
@@ -131,6 +131,13 @@ public RepositoryAuthenticationToken getInstallationToken(String apiUrl, String
                 "No token could be found with access to the requested repository with the given application ID and key");
     }
 
+    private String getV3Url(String apiUrl) {
+        if (apiUrl.matches("^.*/api(/v3)?/?$")) {
+            return apiUrl.replaceFirst("/api(/v3)?/?", "/api/v3");
+        } else {
+            return apiUrl.replaceFirst("/$", "");
+        }
```

Comment:
I'm not a fan of the regular expressions. Could we do this in 2 steps:

```suggestion
        if (apiUrl.endWith("/") {
            apiUrl = apiUrl.substring(0, apiUrl.length() - 1);
        }
        
        if (apiUrl.endsWith("/api/v3")) {
            apiUrl = apiUrl.substring(0, apiUrl.lengh() - 7);
        }
        
        return apiUrl;
```

Reply:
Sure, will do.